### DISCUSSION TITLE
A specification draft for Pod UIs compatible with CSS recipe integration

### DISCUSSION THREAD

**Discussion by joeitu** [solid/community-server-recipes#11] [Ideas]
It could be useful for Pod UI developers to know what is required to make their UI compatible with CSS.

On top of my head:
 - The Pod UI needs to be a Single Page Application
 - It should handle routing properly
 - its routing prefix should be configurable ( thinking of #5 )
 - Optionally, it should have a blank page in order to inject the registration page ( see #6 )

What else is missing?

**Comment by joachimvh** [comment]
I'm not sure there is a straightforward answer to this question. Some options might make things easier for you, but they might not be strict requirements. For example, you say it needs to be a single page application, but you could set up a CSS instance that uses different HTML interfaces based on the path that is being used, it would just require more configuration work.

You could even write a custom module that can be added to the CSS configuration to handle even more specific requirements. This is pretty much what the `ConstantConverter` is, and perhaps this module would have actually made more sense in an external repository to clarify this.

The way the recipes support the user interfaces is just one of the ways it can be done.
